apps/ledger-live-mobile/src/screens/SyncOnboarding/EarlySecurityCheck.tsx

```diff
@@ -1,42 +1,657 @@
+import { track } from "../../analytics";
+import { useGenuineCheck } from "@ledgerhq/live-common/hw/hooks/useGenuineCheck";
+import { useGetLatestAvailableFirmware } from "@ledgerhq/live-common/deviceSDK/hooks/useGetLatestAvailableFirmware";
+import FirmwareUpdateAvailableDrawer from "./FirmwareUpdateAvailableDrawer";
+import { Linking, ScrollView } from "react-native";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { LanguagePrompt } from "./LanguagePrompt";
+import { NavigatorName, ScreenName } from "../../const";
+import { StackNavigationProp } from "@react-navigation/stack";
+import type { UpdateStep } from "../FirmwareUpdate";
+import { urls } from "../../config/urls";
 
-// Only while the ESC implementation is missing
-/* eslint-disable i18next/no-literal-string */
+const LOCKED_DEVICE_TIMEOUT_MS = 1000;
+
+// Represents the UI status of each check step, used by CheckCard
+type UiCheckStatus =
+  | "inactive"
+  | "active"
+  | "completed"
+  | "genuineCheckRefused"
+  | "firmwareUpdateRefused"
+  | "error";
+
+// Represents the status of the genuine check from which is derived the displayed UI and if the genuine check hook can be started or not
+type GenuineCheckStatus = "unchecked" | "ongoing" | "completed" | "error" | "non-genuine";
+// Defines which drawer should be displayed during the genuine check
+type GenuineCheckUiDrawerStatus = "none" | "allow-manager" | "non-genuine" | "genuine-check-error";
+
+// Represents the status of the firmware check from which is derived the displayed UI and if the genuine check hook can be started or not
+type FirmwareUpdateCheckStatus =
+  | "unchecked"
+  | "ongoing"
+  | "updating"
+  | "completed"
+  | "error"
+  | "refused";
+// Defines which drawer should be displayed during the firmware check
+type FirmwareUpdateUiDrawerStatus = "none" | "new-firmware-available";
 
 export type EarlySecurityCheckProps = {
   /**
    * A `Device` object
    */
   device: Device;
+
   /**
    * Function called once the ESC step is finished
    */
   notifyOnboardingEarlyCheckEnded: () => void;
+
+  /**
+   * Called when the device is not in a correct state anymore, for ex when a firmware update has completed and the device probably restarted
+   */
+  notifyEarlySecurityCheckShouldReset: (currentState: { isAlreadyGenuine: boolean }) => void;
+
+  /**
+   * To tell the ESC that there is no need to do a genuine check (optional)
+   *
+   * This will bypass the (idle and) genuine check step and go directly to the firmware update check.
+   * Only useful when the EarlySecurityCheck component is mounting.
+   */
+  isAlreadyGenuine?: boolean;
+
+  /**
+   * Function to cancel the onboarding
+   */
+  onCancelOnboarding: () => void;
 };
 
 /**
  * Component representing the early security checks step, which polls the current device state
  * to display correctly information about the onboarding to the user
  */
 export const EarlySecurityCheck: React.FC<EarlySecurityCheckProps> = ({
```

Comment:
can we split this huge component in a `ComponentA` handling the logic and a `ComponentB` with 0 logic and just rendering stuff?
similar to here https://github.com/LedgerHQ/ledger-live/pull/3909/files#diff-dff85510b1a6ee2a82f5094b9dc0cb7298e1dd8c1a2a3f3ff62bee26e1bf16ab

Reply:
Done ;)